imports/plugins/core/core/server/method-hooks.js

```diff
@@ -74,21 +74,20 @@ MethodHooks._initializeHook = function (mapping, methodName, hookFunction) {
     // Get arguments you can mutate
     const args = _.toArray(inputArgs);
     let beforeResult;
-    // Call the before hooks
+    let hooksProcessed = 0;
 
-    const beforeHooks = MethodHooks._beforeHooks[methodName];
-    _.each(beforeHooks, (beforeHook, hooksProcessed) => {
-      beforeResult = beforeHook.call(this, {
+    // Call the before hooks
+    const beforeHooks = MethodHooks._beforeHooks[methodName] || [];
+    for (const beforeHook of beforeHooks) {
+      beforeResult = Promise.await(beforeHook.call(this, {
         result: undefined,
         error: undefined,
         arguments: args,
         hooksProcessed
-      });
+      }));
 
-      if (beforeResult === false) {
```

Comment:
Probably better to keep this early return here and remove the one after the loop, assuming that returning `false` is supposed to be a way to stop the remaining hooks from running. (Not sure if that's the case, or if any hooks actually use this.)